d4/src/ssio/reader.rs

```diff
@@ -147,6 +149,8 @@ impl<R: Read + Seek> D4TrackReader<R> {
                             ));
                         }
                     }
+                } else {
+                    break;
```

Comment:
I think this introduce new bug. Basically this is the part that creates a view to secondary table partitions that contains the data points that is related to the query.

If break at this point, I believe this will introduce bug in some query for large interval which is larger than a secondary table partition.

So I am a little bit confused if the issue is just change the interval convention, what is the problem that needs to be fixed at this point. 

To help me understand the change better, could you please give me some data that triggers a buggy output. 

Thanks,
Hao

Reply:
Hi @38, in case it helps I link to data that appears to be bugged in issue #59 and walk through the example.